libr/util/print.c

```diff
@@ -1564,9 +1565,44 @@ R_API void r_print_progressbar(RPrint *p, int pc, int _cols) {
 	p->cb_printf ("]");
 }
 
+/* TODO: handle screen width */
+R_API void r_print_progressbar_with_count(RPrint *p, int pc, int total, int _cols, bool reset_line) {
+	int i, cols = (_cols == -1)? 78: _cols;
+	if (!p) {
+		p = &staticp;
+	}
+	const bool enable_colors = p && (p->flags & R_PRINT_FLAGS_COLOR);
+	const char *h_line = p->cons->use_utf8? RUNE_LONG_LINE_HORIZ: "-";
+	const char *block = p->cons->use_utf8? UTF_BLOCK: "#";
+
+	pc = R_MAX (0, R_MIN (total, pc));
+	if (reset_line) {
+		p->cb_printf ("\r");
+	}
+	if (p->flags & R_PRINT_FLAGS_HEADER) {
+		if (enable_colors) {
+			p->cb_printf ("%s%4d%s%% %s%6d%s/%6d ", Color_GREEN, pc * 100 / total, Color_RESET, Color_GREEN, pc, Color_RESET, total);
+		} else {
+			p->cb_printf ("%4d%% %6d/%6d ", pc * 100 / total, pc, total);
+		}
+	}
+	cols -= 29;
+	enable_colors? p->cb_printf ("[%s", Color_YELLOW): p->cb_printf ("[");
+	for (i = cols * pc / total; i; i--) {
```

Comment:
uhm there are two bugs in this division:

* total can be 0 and cause divbyzero exception
* pc is signed, so a division unrepresentable result can happen too

i wrote some macros few commits ago in master to detect such issues, use that before entering the loop. and add proper checks to avoid negative values in in cols/total/..

Reply:
I've made most of the requested changes, but I'm a little unsure about what to do for this. How exactly should I handle these cases? I'm not sure what macros you're referring to.

Reply:
I've attempted to handle these cases adequately. The only macros I found were in `r_types_overflow.h`, and it wasn't clear to me how they would be used in this context.